testutil/keeper/expectations.go

```diff
@@ -217,3 +217,36 @@ func GetMocksForSlashValidator(
 			Times(1),
 	}
 }
+
+// SetupMocksForLastBondedValidatorsExpectation sets up the expectation for the `IterateLastValidatorPowers` `MaxValidators`, and `GetValidator` methods of the `mockStakingKeeper` object.
+// These are needed in particular when calling `GetLastBondedValidators` from the provider keeper.
+// Times is the number of times the expectation should be called. Provide -1 for `AnyTimes“.
+func SetupMocksForLastBondedValidatorsExpectation(mockStakingKeeper *MockStakingKeeper, maxValidators uint32, vals []stakingtypes.Validator, powers []int64, times int) {
```

Comment:
Open to suggestions for avoiding/improving the times parameter. I think it's not too bad, but not very pretty code, either